### PR TITLE
Start explaining parameters

### PR DESCRIPTION
**Pull request by PythonCoderAS**
Fancy tables and a new model reference.

### CODE REVIEW

File: src/help/guides/extension-development/cheerio-quickstart.md [hunk 1]
```diff
@@ -23,7 +24,9 @@ $('div.apple').text()	// Gets the text of the first 'div' block which ALSO has c
 <p id="someText" lang="en">
     This is some text
 </p>
+```
 ------------
```
Comment (FramboisePi) [inline src/help/guides/extension-development/cheerio-quickstart.md:28]: We usually use three `-`
```suggestion
---
```

File: src/help/guides/extension-development/function-definitions.md [hunk 2]
```diff
@@ -1,13 +1,18 @@
 # Required Methods
 ## getMangaDetails
+### Method signature
 `async getMangaDetails(mangaId: string): Promise<Manga>`
 
-- A manga ID is defined by the source developer. However IDs are provided to Paperback in other functions, will be given to this function in the same format.
+### Parameters
+| Parameter | Type | Description|
+|-----------|------|------------|
+|  mangaId  | String | The ID of a manga. The manga ID is provided from the other discovery functions, such as the home page and performed searches. The manga ID should be used to return information about the manga, such as the title and author.|
```
Comment (FramboisePi) [inline src/help/guides/extension-development/function-definitions.md:9]: What would you think of using a code bloc? I'm not sure if its better with or without
```suggestion
|  `mangaId`  | String | The ID of a manga. The manga ID is provided from the other discovery functions, such as the home page and performed searches. The manga ID should be used to return information about the manga, such as the title and author.|
```

File: src/help/guides/extension-development/function-definitions.md [hunk 3]
```diff
@@ -1,13 +1,18 @@
...
 
-Given an ID, this async function should return a filled out Manga object.
+### Returns
+Given an ID, this async function should return a filled out [Manga](model-reference.md#Manga) object.
```
Comment (FramboisePi) [inline src/help/guides/extension-development/function-definitions.md:12]: We usually don't use file extension in links. Should we?
```suggestion
Given an ID, this async function should return a filled out [Manga](model-reference/#Manga) object.
```

File: src/help/guides/extension-development/function-definitions.md [hunk 4]
```diff
@@ -87,14 +98,19 @@ async getChapters(mangaId: string): Promise<Chapter[]> {
...
 ## getChapterDetails
+### Method Signature
 `async getChapterDetails(mangaId: string, chapterId: string): Promise<ChapterDetails>`
 
-- A manga ID is defined by the source developer. However IDs are provided to Paperback in other functions, will be given to this function in the same format.
-- A ChapterID is defined by the [Get Chapters](#getChapters) method. However it is returned there, will be fed into this function.
+### Parameters
+| Parameter | Type | Description|
+|-----------|------|------------|
+|  mangaId  | String | The ID of a manga. The manga ID is provided from the other discovery functions, such as the home page and performed searches. The manga ID should be used to return information about the manga, such as the title and author.|
```
Comment (FramboisePi) [inline src/help/guides/extension-development/function-definitions.md:107]:
```suggestion
|  mangaId  | String | The ID of a manga. The manga ID is provided from the other discovery functions, such as the home page and performed searches. The manga ID should be used to return information about the manga, such as the title and author. |
```

File: package.json [hunk 5]
```diff
@@ -22,7 +22,7 @@
 		"eslint-config-vuepress": "^2.1.0",
 		"eslint-plugin-prettier": "^3.1.4",
 		"prettier": "^2.0.5",
-		"vuepress": "^1.5.2",
+		"vuepress": "^1.8.2",
```
Comment (FramboisePi) [inline package.json:25]: I believe we may prefer to keep dependencies update in a separate pull request, maybe to the `master` branch? Unless there is a need of this update for this pull request. This would reduce a bit the amount of things that need to be tested.

File: src/help/guides/extension-development/intro-to-sources.md [hunk 6]
```diff
@@ -1,12 +1,23 @@
 ### Introduction: A Manga Reader Without Servers
-When developing a Manga Reader, first and foremost, you need to gather your content from somewhere. There are many services out there, such as WebToons or MangaDex which run their own content providing system which their applications can attach to without much issues.
 
-If you do not hold this kind of infrastructure, as Paperback does not, the app needs to be designed in a way where it can dynamically read **other** content providers.
+When developing a Manga Reader, first and foremost, you need to gather your content from somewhere. There are many
+services out there, such as WebToons or MangaDex which run their own content providing system which their applications
```
Comment (FramboisePi) [inline src/help/guides/extension-development/intro-to-sources.md:4]: I'm not sure to what refer "WebToons", could it be "WEBTOON"?

File: src/help/guides/extension-development/intro-to-sources.md [hunk 7]
```diff
@@ -1,12 +1,23 @@
...
+can attach to without many issues.
 
-This design paradigm has many benefits. The Apple Store is very restrictive on content provided to iOS devices, however with a generic source system which can be added into the application at runtime allows all sorts of content to be made available, only requiring the users to install each module correctly.
+If you do not hold this kind of infrastructure, as Paperback does not, the app needs to be designed in a way where it
+can dynamically read **other** content providers.
 
-Enter the Paperback Extension system. If you are familiar with [Tachiyomi](https://tachiyomi.org) this system is practically the same. Users are free to develop an extension completely independant with the Paperback application builds and updates, and publish generic sources which are simply parsing definitions which understand how to import the contents of X website.
+This design paradigm has many benefits. The Apple Store is very restrictive on content provided to iOS devices, however
```
Comment (FramboisePi) [inline src/help/guides/extension-development/intro-to-sources.md:10]:
```suggestion
This design paradigm has many benefits. The Apple App Store is very restrictive on content provided to iOS and iPadOS devices, however
```

File: src/help/guides/extension-development/metadata.md [hunk 8]
```diff
@@ -0,0 +1,48 @@
...
+		method: GET,
+		...
+	});
+    /* Do request and parse data */
+	return createPagedResults{
+	    results: results,
+		metadata: {
+	        page: page
+		}
+	}
```
Comment (FramboisePi) [inline src/help/guides/extension-development/metadata.md:46]: The indentation seems a bit off
```suggestion
    if (typeof metadata === "object" && metadata.page){
        page = metadata.page;
    } else {
        page = 1;
    }
    const request = createRequest({
        url: `https://mysite.com/search/?query=${query.title}&page=${page}`,
        method: GET,
        ...
    });
    /* Do request and parse data */
    return createPagedResults{
        results: results,
        metadata: {
            page: page
        }
    }
```

File: src/help/guides/extension-development/metadata.md [hunk 9]
```diff
@@ -0,0 +1,48 @@
...
+    let page;
+    if (typeof metadata === "object" && metadata.page){
+        page = metadata.page;
+	} else {
+        page = 1;
+	}
+    const request = createRequest({
+		url: `https://mysite.com/search/?query=${query.title}&page=${page}`,
+		method: GET,
+		...
```
Comment (FramboisePi) [inline src/help/guides/extension-development/metadata.md:38]: We may want to suggest the use of `param` for the title and the page
```suggestion
        url: `https://mysite.com/search/`,
        param: `?query=${query.title}&page=${page}`
        method: GET,
        ...
```

File: src/help/guides/extension-development/metadata.md [hunk 10]
```diff
@@ -0,0 +1,48 @@
...
+	});
+    /* Do request and parse data */
```
Comment (FramboisePi) [inline src/help/guides/extension-development/metadata.md:40]: I think the rest of the guide use `//`
```suggestion
    // Do request and parse data
```

File: src/help/guides/extension-development/model-reference.md [hunk 11]
```diff
@@ -0,0 +1,161 @@
...
+| id | String | The chapter ID of the chapter that the object respresents. |
+| mangaId | String | The given identifier of the Manga that owns this chapter. This should match the id of the manga that the chapter is for. |
+| pages | String Array | The list of pages in the chapter. The pages are ordered by the index of the array, where the 0th index is the first page, and so on. Each item should be a link to the image file that represents the page. |
+| longStrip | Boolean | Indicates whether or not the chapter should be rendered in **Long Strip Mode**. A long strip is a webtoon or any other form of manga that uses very tall but narrow images. |
+
+## LanguageCode
+
+A string enum defining full language names, and the corresponding language code.
+
+:::warning
```
Comment (FramboisePi) [inline src/help/guides/extension-development/model-reference.md:44]: You can put a title for the warning block. We usually use "Warning"
```suggestion
::: warning Warning
```

File: src/help/guides/extension-development/intro-to-sources.md [hunk 12]
```diff
@@ -1,12 +1,23 @@
 ### Introduction: A Manga Reader Without Servers
-When developing a Manga Reader, first and foremost, you need to gather your content from somewhere. There are many services out there, such as WebToons or MangaDex which run their own content providing system which their applications can attach to without much issues.
 
-If you do not hold this kind of infrastructure, as Paperback does not, the app needs to be designed in a way where it can dynamically read **other** content providers.
+When developing a Manga Reader, first and foremost, you need to gather your content from somewhere. There are many
+services out there, such as WebToons or MangaDex which run their own content providing system which their applications
+can attach to without many issues.
```
Comment (FramboisePi) [inline src/help/guides/extension-development/intro-to-sources.md:5]: Do we want to cut this text in multiple lines? It is something not done on the rest of the website

**Review by FramboisePi — CHANGES_REQUESTED** [review]
First off all, thank you PokestarFan for this big pull request.
As there is a lot of modifications, the review process may take some time. 
I will only review the form but let other extension developers give their feedback on the content.

Some files may need to be moved as it is planned to change the folders organization in order to create a contribution section. I believe we can keep things as they are for now and move them latter.

File: src/.vuepress/config/sidebar/en.js [hunk 13]
```diff
@@ -14,6 +14,12 @@ module.exports = [
 	{
 		title: "Source Development",
 		collapsable: true,
-		children: ["/help/guides/extension-development/intro-to-sources", "/help/guides/extension-development/cheerio-quickstart", "/help/guides/extension-development/function-definitions"]
+		children: [
+			"/help/guides/extension-development/intro-to-sources",
+			"/help/guides/extension-development/cheerio-quickstart",
+			"/help/guides/extension-development/metadata/",
```
Comment (FramboisePi) [inline src/.vuepress/config/sidebar/en.js:20]:
```suggestion
			"/help/guides/extension-development/metadata",
```

**Review by FramboisePi — CHANGES_REQUESTED** [review]

File: package.json [hunk 14]
```diff
@@ -35,6 +35,8 @@
 		"eslint-plugin-react": "^7.20.3",
 		"eslint-plugin-react-hooks": "^2.5.1",
 		"prettier-config-vuepress": "^1.0.0",
+		"vue": "^2.6.12",
+		"vue-server-renderer": "^2.6.12",
```
[the remark this answers is not in the document]
Comment (PythonCoderAS) [reply package.json:39]: When I did an upgrade, they got out of sync, so I had to lock them both to the same version.

File: src/help/guides/extension-development/function-definitions.md (hunk 3 above)
Comment (PythonCoderAS) [reply src/help/guides/extension-development/function-definitions.md:12]: It doesn't make a difference from what I've seen.

File: src/help/guides/extension-development/function-definitions.md (hunk 2 above)
Comment (PythonCoderAS) [reply src/help/guides/extension-development/function-definitions.md:9]: Should this be done for every parameter? I mean it would help distinguish between uppercase I's and lowercase l's.

File: package.json (hunk 5 above)
Comment (PythonCoderAS) [reply package.json:25]: I tried to upgrade the version to fix the sidebar, which did not help. I mostly did so because vuepress github required latest version before filing an issue.

File: src/help/guides/extension-development/intro-to-sources.md (hunk 6 above)
Comment (PythonCoderAS) [reply src/help/guides/extension-development/intro-to-sources.md:4]: WebToons are a type of long strip content made for viewing on mobile devices. WEBTOON is one company that publishes them.

Comment (PythonCoderAS) [reply src/help/guides/extension-development/intro-to-sources.md:4]: Also, I did not modify this, just formatted it, so the text is from the old GitHub wiki.

File: src/help/guides/extension-development/metadata.md (hunk 8 above)
Comment (PythonCoderAS) [reply src/help/guides/extension-development/metadata.md:46]: Hmm, my editor must have been messing with the indentation, and I also thought it was problematic.

File: src/help/guides/extension-development/metadata.md (hunk 9 above)
Comment (PythonCoderAS) [reply src/help/guides/extension-development/metadata.md]: Param does a direct URL concatenation, so I don't see the point in using it. I would use param if I could provide it an object, but until it does so, there is no difference.

File: src/help/guides/extension-development/intro-to-sources.md (hunk 12 above)
Comment (PythonCoderAS) [reply src/help/guides/extension-development/intro-to-sources.md:5]: I planned on doing so for the rest of the website, as markdown only considers a double newline a new paragraph. It does help editing files if you don't have wrapping on.

File: src/help/guides/extension-development/metadata.md (hunk 9 above)
Comment (ConradWeiser) [reply src/help/guides/extension-development/metadata.md]: I'm fine with this. I've never really liked the parameter value, supplying the parameters directly into the URL I'm fine with being the documented method for this sort of thing.

File: src/help/guides/extension-development/intro-to-sources.md (hunk 12 above)
Comment (ConradWeiser) [reply src/help/guides/extension-development/intro-to-sources.md:5]: Keeping it consistent is a good call. I see that you **are** keeping it consistent with the multiple lines through this PR. I'm a little unsure how I feel about it, much preferring the one-line style letting your IDE do the wrapping, but this isn't a dealbreaker I guess.

File: src/help/guides/extension-development/function-definitions.md [hunk 15]
```diff
@@ -1,13 +1,24 @@
...
+
 `async getMangaDetails(mangaId: string): Promise<Manga>`
 
-- A manga ID is defined by the source developer. However IDs are provided to Paperback in other functions, will be given to this function in the same format.
+### Parameters
 
-Given an ID, this async function should return a filled out Manga object.
+| Parameter | Type | Description|
+|-----------|------|------------|
+|  `mangaId`  | String | The ID of a manga. The manga ID is provided from the other discovery functions, such as the home page and performed searches. The manga ID should be used to return information about the manga, such as the title and author.|
```
Comment (ConradWeiser) [inline src/help/guides/extension-development/function-definitions.md:13]: Be specific with the function names. Even better, link them!

File: src/help/guides/extension-development/function-definitions.md [hunk 16]
```diff
@@ -127,6 +159,76 @@ async getChapterDetails(mangaId: string, chapterId: string): Promise<ChapterDeta
...
+		...
+	});
+	/*
+	Execute the request, and retrieve the HTML page
+	this.requestManager is provided to you by the parent class. Always use this to make requests.
+	The second parameter is the number of retries the app is allowed to make, if the request fails the first time.
+	*/
+	const data = await this.requestManager.schedule(request, 1)
+
+	// Prepare to parse the page using CheerioJS (Class object included by parent class)
```
Comment (ConradWeiser) [inline src/help/guides/extension-development/function-definitions.md:203]:
```suggestion
	// Prepare to parse the page using CheerioJS (The cheerio object is provided by the base class this extends)
```
I think this was a copy and paste from my original draft, lets do a slight update

File: src/help/guides/extension-development/model-reference.md [hunk 17]
```diff
@@ -0,0 +1,269 @@
...
+| `icon` | String | THe icon that the object represents. |
+
+## LanguageCode
+
+A string enum defining full language names, and the corresponding language code.
+
+::: warning Warning Some full language names have typos. These may be fixed in a future version.
+:::
+
+### Values
```
Comment (ConradWeiser) [inline src/help/guides/extension-development/model-reference.md:94]: Do we really want to include these? Letting them know that the language code exists is nice. Is giving all of the possible values a waste of space? Any good IDE will list out the enum as soon as they try to use it.

If you think this is fine for readability, keeping it I guess is fine? Just something to consider.

File: src/help/guides/extension-development/model-reference.md [hunk 18]
```diff
@@ -0,0 +1,269 @@
...
+source to decide. These objects use the [metadata](metadata.md) field.
+
+### Required Fields
+
+| Name | Type | Description |
+|------|------|-------------|
+| results | [MangaTile](#mangatile) Array | The mangas that were found in the page. |
+| metadata | any | The metadata to carry onto future calls of the function returning the object. Reference the [page on metadata](metadata.md) to learn more. |
+
+## SearchRequest
```
Comment (ConradWeiser) [inline src/help/guides/extension-development/model-reference.md:217]: Emphasize that `title` is the only field that is supported at the moment.
With the current development direction, likely everything else will be removed from this method. An Advanced Search method will have different fields.

**Review by ConradWeiser — COMMENTED** [review]
As an overall comment - I really am not fond of splitting up sentences/paragraphs to multiple lines like that. But that's an unseen semantic, as long as everything else follows this standard, that's fine. 

The display of required and optional fields for all of the methods is very informative!

Overall, it looks pretty solid!

File: src/help/guides/extension-development/function-definitions.md (hunk 15 above)
Comment (PythonCoderAS) [reply src/help/guides/extension-development/function-definitions.md:13]: The functions aren't added, but I will link them once I do add them in.

File: src/help/guides/extension-development/model-reference.md (hunk 17 above)
Comment (PythonCoderAS) [reply src/help/guides/extension-development/model-reference.md:94]: The model reference is supposed to be a model reference, meaning that all model values are listed. Also god forbid someone uses notepad++ or vim to fix a typo. It would be nice if this was collapsible though.

File: src/help/guides/extension-development/model-reference.md (hunk 18 above)
Comment (PythonCoderAS) [reply src/help/guides/extension-development/model-reference.md:217]: I thought it was mentioned that Paperback only supports the title.

File: src/help/guides/extension-development/model-reference.md (hunk 17 above)
Comment (FramboisePi) [reply src/help/guides/extension-development/model-reference.md:94]: For a collapsible component, you can use
```markdown
:::: el-collapse
::: el-collapse-item title="Lorem ipsum"
**Lorem ipsum** dolor sit amet, _consectetur_ adipiscing **elit**.
:::
::: el-collapse-item title="Pellentesque"
**Pellentesque** _habitant_ morbi tristique **senectus** et netus.
:::
::::
```

File: src/help/guides/extension-development/intro-to-sources.md (hunk 6 above)
Comment (FramboisePi) [reply src/help/guides/extension-development/intro-to-sources.md:4]: @ConradWeiser when you wrote "WebToons", were you talking about the format or the company? The presence of MangaDex let me suppose it was the company but I could be mistaking.

File: src/help/guides/extension-development/cheerio-quickstart.md [hunk 19]
```diff
@@ -157,12 +162,15 @@ With a given HTML document, lets do some parsing.
...
 ```
 
-Given the above set of HTML, say that you have loaded it into Cheerio using the instructions above. What you are looking for, is the text inside each of the `li` objects. There should be a few thoughts which should be going through your head;
+Given the above set of HTML, say that you have loaded it into Cheerio using the instructions above. What you are looking
+for, is the text inside each of the `li` objects. There should be a few thoughts which should be going through your
+head;
 
 - What _unique_ identifier is surrounding the data that I need?
-  - In the example of parsing 'Three Apples', we can see that it is __always__ in a `li` block, with a class of `apple`
+	- In the example of parsing 'Three Apples', we can see that it is __always__ in a `li` block, with a class
```
Comment (FramboisePi) [inline src/help/guides/extension-development/cheerio-quickstart.md:170]: This could be a consequences of the linting but we used to use `*` instead of `-`. I'm not against `-` but I propose we chose which we want to use.
```suggestion
	- In the example of parsing 'Three Apples', we can see that it is **always** in a `li` block, with a class
```

File: src/help/guides/extension-development/cheerio-quickstart.md [hunk 20]
```diff
@@ -171,17 +179,17 @@ $('li.apple')		// Selects the correct spot
 $('li.apple').text()	// This will return the text inside of the selected area.
 ```
 
-What if we want to get the "Twelve Pears" text? Referencing the table at the top of this document, we can use the **Attribute Equals Selector**!
+What if we want to get the "Twelve Pears" text? Referencing the table at the top of this document, we can use the **
+Attribute Equals Selector**!
```
Comment (FramboisePi) [inline src/help/guides/extension-development/cheerio-quickstart.md:183]: The new line breaks the markdown
```suggestion
What if we want to get the "Twelve Pears" text? Referencing the table at the top of this document, we can use the **Attribute Equals Selector**!
```

File: src/help/guides/extension-development/function-definitions.md [hunk 21]
```diff
@@ -82,19 +103,30 @@ async getChapters(mangaId: string): Promise<Chapter[]> {
...
+
+| Parameter | Type | Description|
+|-----------|------|------------|
+|  `mangaId`  | String | The ID of a manga. The manga ID is provided from the other discovery functions, such as the home page and performed searches. The manga ID should be used to return information about the manga, such as the title and author. |
+| `chapterId` | String | The ID of a chapter. defined by the [Get Chapters](#getchapters) method. The chapter ID can only contain letters, numbers, dashes and underscores. |
 
-Given both a Manga and a Chapter ID, this function should populate a ChapterDetails object, which goes into further detail of what each chapter contains.
+### Returns
+
+Given both a Manga and a Chapter ID, this function should populate a [ChapterDetails](model-reference.md#ChapterDetails)
```
Comment (FramboisePi) [inline src/help/guides/extension-development/function-definitions.md:125]: As you said there was no difference, I would prefer:
```suggestion
Given both a Manga and a Chapter ID, this function should populate a [ChapterDetails](model-reference/#ChapterDetails)
```

File: src/help/guides/extension-development/function-definitions.md [hunk 22]
```diff
@@ -127,6 +159,76 @@ async getChapterDetails(mangaId: string, chapterId: string): Promise<ChapterDeta
...
+
+### Method signature
+
+`async searchRequest(query: SearchRequest, metadata: any): Promise<PagedResults>`
+
+### Parameters
+
+| Parameter | Type | Description|
+|-----------|------|------------|
+| `query` | [SearchRequest](model-reference.md#searchrequest) | The object representing the search. View the fields of the [SearchRequest](model-reference.md#searchrequest) object to find out what a search can contain. |
```
Comment (FramboisePi) [inline src/help/guides/extension-development/function-definitions.md:172]:
```suggestion
| `query` | [SearchRequest](model-reference/#searchrequest) | The object representing the search. View the fields of the [SearchRequest](model-reference/#searchrequest) object to find out what a search can contain. |
```

File: src/help/guides/extension-development/function-definitions.md [hunk 23]
```diff
@@ -127,6 +159,76 @@ async getChapterDetails(mangaId: string, chapterId: string): Promise<ChapterDeta
...
+| `metadata` | any | Metadata carried through function calls. View the [metadata page](metadata.md) to learn more about how the parameter works. |
```
Comment (FramboisePi) [inline src/help/guides/extension-development/function-definitions.md:173]:
```suggestion
| `metadata` | any | Metadata carried through function calls. View the [metadata page](metadata/) to learn more about how the parameter works. |
```

File: src/help/guides/extension-development/function-definitions.md [hunk 24]
```diff
@@ -127,6 +159,76 @@ async getChapterDetails(mangaId: string, chapterId: string): Promise<ChapterDeta
...
+
+### Returns
+
+A [PagedResults](model-reference.md#pagedresults) object with the results of the search for the current page, and metadata for the next page, if it exists.
```
Comment (FramboisePi) [inline src/help/guides/extension-development/function-definitions.md:177]:
```suggestion
A [PagedResults](model-reference/#pagedresults) object with the results of the search for the current page, and metadata for the next page, if it exists.
```

File: src/help/guides/extension-development/intro-to-sources.md (hunk 6 above)
Comment (ConradWeiser) [reply src/help/guides/extension-development/intro-to-sources.md:4]: Originally it was the company/group, yeah.

Comment (FramboisePi) [reply src/help/guides/extension-development/intro-to-sources.md:4]: Thanks Vibrant. Then I propose to use
```suggestion
services out there, such as WEBTOON or MangaDex which run their own content providing system which their applications
```

File: src/help/guides/extension-development/function-definitions.md (hunk 21 above)
Comment (PythonCoderAS) [reply src/help/guides/extension-development/function-definitions.md:125]: Actually, while I did say there was no difference, the latter breaks on my local testing system, so I actually think the first one is the way to go.

File: src/help/guides/extension-development/function-definitions.md (hunk 22 above)
Comment (PythonCoderAS) [reply src/help/guides/extension-development/function-definitions.md:172]: per above

File: src/help/guides/extension-development/function-definitions.md (hunk 23 above)
Comment (PythonCoderAS) [reply src/help/guides/extension-development/function-definitions.md:173]: Per above

File: src/help/guides/extension-development/function-definitions.md (hunk 24 above)
Comment (PythonCoderAS) [reply src/help/guides/extension-development/function-definitions.md:177]: Per above

File: src/help/guides/extension-development/function-definitions.md [hunk 25]
```diff
@@ -127,6 +159,76 @@ async getChapterDetails(mangaId: string, chapterId: string): Promise<ChapterDeta
...
+	if (typeof metadata === "object" && metadata.page) {
+		page = metadata.page;
+	} else { // If there is no current page, this must mean this is the first page.
+		page = 1;
+	}
+	const request = createRequest({
+		url: `https://mysite.com/search/?query=${query.title}&page=${page}`,
+		method: GET,
+		...
+	});
```
Comment (FramboisePi) [inline src/help/guides/extension-development/function-definitions.md:195]: Should it be `createRequestObject`?
```suggestion
	const request = createRequestObject({
		url: `https://mysite.com/search/?query=${query.title}&page=${page}`,
		method: GET,
		...
	});
```

File: src/help/guides/extension-development/model-reference.md [hunk 26]
```diff
@@ -0,0 +1,267 @@
...
+| `lastUpdate` | String | A string showing when the manga was last updated. It might either be an absolute date (`March 1, 2021`) or a relative date (`2 days ago`). |
+
+## MangaStatus
+
+An integer enum representing the various statuses a manga can have.
+
+### Values
+
+| Name | Value | Description |
+|------|-------|-------------|
```
Comment (FramboisePi) [inline src/help/guides/extension-development/model-reference.md:180]: As the column contains only integers, what would you think of centering the value? That would be inconsistent with the rest of the page so I'm not sure if its a good idea.
```suggestion
|------|:-----:|-------------|
```

Comment (PythonCoderAS) [reply src/help/guides/extension-development/model-reference.md:180]: There's no point, since the number column is very tiny relatively.

Comment (ConradWeiser) [reply src/help/guides/extension-development/model-reference.md:180]: I like the idea of centering it. It's a solid style guideline.

Comment (PythonCoderAS) [reply src/help/guides/extension-development/model-reference.md:180]: But why this column? In that case, shouldn't it apply to all enums?

Comment (ConradWeiser) [reply src/help/guides/extension-development/model-reference.md:180]: It should apply to columns which __only__ have integers. If the enumeration mapping is to strings or some other type, this wouldn't fit.

Comment (PythonCoderAS) [reply src/help/guides/extension-development/model-reference.md:180]: I guess it makes sense with numbers.

File: src/help/guides/extension-development/function-definitions.md (hunk 21 above)
Comment (ConradWeiser) [reply src/help/guides/extension-development/function-definitions.md:125]: I'm having better luck with what @FramboisePi has suggested. How does this break your local system?
I'm getting a strange navigation frame center with what's currently present here.

File: src/help/guides/extension-development/model-reference.md (hunk 17 above)
Comment (ConradWeiser) [reply src/help/guides/extension-development/model-reference.md:94]: Given how frequently instances of a word is also a navigation link (Ex: ChapterDetails usually sends you to the ChapterDetails page) - I suppose this really isn't all too much of a problem.
To keep consistency, there's really no need to collapse this. It's not like the only way a user can navigate this page is by scrolling and looking for things manually.

I don't think I mind how this is currently, and am alright with dismissing this comment.

File: src/help/guides/extension-development/function-definitions.md (hunk 21 above)
Comment (PythonCoderAS) [reply src/help/guides/extension-development/function-definitions.md:125]: For some reason, it's working now. I may have made a typo earlier causing it to not work, but regardless it seems to work now.

File: src/help/guides/extension-development/model-reference.md (hunk 17 above)
Comment (PythonCoderAS) [reply src/help/guides/extension-development/model-reference.md:94]: That seems fair.

File: src/help/guides/extension-development/model-reference.md (hunk 18 above)
Comment (PythonCoderAS) [reply src/help/guides/extension-development/model-reference.md:217]: I think this is fair for now because it's only a week or so until the stateful update comes in and this needs to be rewritten anyways.

**Review by FramboisePi — APPROVED** [review]
Looks good to me. Thank you for this important work!

**Review by ConradWeiser — APPROVED** [review]
